### PR TITLE
feat(deno): Retrieve version from import.meta.url or git, when possible

### PR DESCRIPTION
**Pull request by effigies**
```
❯ deno run --reload --allow-all https://github.com/effigies/bids-validator/raw/enh/deno-version/bids-validator/src/bids-validator.ts --version
bids-validator enh/deno-version
❯ deno run --allow-all bids-validator/src/bids-validator.ts --version
bids-validator v1.14.7-dev.0-38-g5314ea00
```

Should emit the right version for released versions on deno.land. We can improve it for untagged versions in the future.

### CODE REVIEW

**Comment by effigies** [comment]
Bummer...

```
x Build failed in 2.07s
error during build:
Error: [vite:esbuild-transpile] Transform failed with 1 error:
assets/index-!~{001}~.js:5620:43: ERROR: Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)

Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
5618|  new Command().name("bids-validator").type("debugLevel", new EnumType(LogLevelNames)).description(
5619|    "This tool checks if a dataset in a given directory is compatible with the Brain Imaging Data Structure specification. To learn more about Brain Imaging Data Structure visit http://bids.neuroimaging.io/"
5620|  ).arguments("<dataset_directory>").version(await getVersion()).option("--json", "Output machine readable JSON").option(
   |                                             ^
5621|    "-s, --schema <type:string>",
5622|    "Specify a schema version to use for validation",
```

**Comment by effigies** [comment]
I think this should be good to go.

File: bids-validator/src/version.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * @returns The version of the script.
+ *
+ */
+export async function getVersion(): Promise<string> {
+  // Hard-coded JSON wins
+  let version = getArchiveVersion()
+  if (version) { return version }
+
+  const url = new URL(Deno.mainModule)
```
Comment (effigies) [inline bids-validator/src/version.ts:26]: I switched to `Deno.mainModule` to display a module users would expect, if we can't find any version info. Not sure if tying it to Deno is wise. Could inspect the stack trace to find the main module.

File: bids-validator/src/setup/options.ts [hunk 2]
```diff
@@ -12,34 +13,36 @@ export type ValidatorOptions = {
...
-    'Disregard NIfTI header content during validation',
-  )
-  .option('--debug <type:debugLevel>', 'Enable debug output', {
-    default: 'ERROR',
-  })
-  .option(
-    '--filenameMode',
-    'Enable filename checks for newline separated filenames read from stdin',
-  )
+async function getValidatorCommand() {
```
Comment (effigies) [inline bids-validator/src/setup/options.ts]: The return annotation caused headaches, so I dropped it, and typescript was happy enough.

**Review by nellh — APPROVED** [review]
I made one change which preserves the export for validateCommand since this is used by OpenNeuro CLI.

**Comment by effigies** [comment]
Rebased to resolve tests (I hope).

**Pull request merged**